dbee/adapters/duck_driver.go

```
@@ -2,52 +2,75 @@ package adapters
-	return c.c.ColumnsFromQuery("DESCRIBE %q", opts.Table)
+	return c.c.ColumnsFromQuery("DESCRIBE %q.%q", opts.Schema, opts.Table)
 }
 
 func (c *duckDriver) Structure() ([]*core.Structure, error) {
-	query := `SHOW TABLES;`
+	catalogQuery := fmt.Sprintf(`
+		SELECT table_schema, table_name, table_type
+		FROM information_schema.tables
+		WHERE table_catalog = '%s';`,
+		c.currentCatalog)
 
-	rows, err := c.Query(context.TODO(), query)
+	rows, err := c.Query(context.Background(), catalogQuery)
 	if err != nil {
 		return nil, err
 	}
 
-	var schema []*core.Structure
-	for rows.HasNext() {
-		row, err := rows.Next()
-		if err != nil {
-			return nil, err
-		}
-
-		// We know for a fact there is only one string field (see query above)
-		table := row[0].(string)
-		schema = append(schema, &core.Structure{
-			Name:   table,
-			Schema: "",
-			Type:   core.StructureTypeTable,
-		})
+	return core.GetGenericStructure(rows, getDuckDBStructureType)
+}
+
+// getDuckDBStructureType returns the core.StructureType based on the
+// given type string for duckdb adapter.
+func getDuckDBStructureType(typ string) core.StructureType {
+	// TODO: (phdah) Add more types if exists
+	switch typ {
+	case "BASE TABLE":
+		return core.StructureTypeTable
+	case "VIEW":
+		return core.StructureTypeView
+	default:
+		return core.StructureTypeNone
 	}
+}
+
+// ListDatabases returns the current catalog and a list of available catalogs.
+// NOTE: (phdah) As of now, swapping catalogs is not enabled and only the
+// current will be shown
+func (c *duckDriver) ListDatabases() (current string, available []string, err error) {
+	// no-op
+	return c.currentCatalog, []string{c.currentCatalog}, nil
+}
 
-	return schema, nil
+// SelectDatabase switches the current database/catalog to the selected one.
+// NOTE: (phdah) As of now, swapping catalogs is not enabled
+func (c *duckDriver) SelectDatabase(name string) error {
+	// no-op
+	return nil
```

let's return an error here of `fmt.Errorf("SelectDabase method not implemented yet")` or something similar. Otherwise, it might be easy to confuse the user that there is a no-op even tho I am selecting a new database 'succesfully'. wdyt? 😄

---

Sounds good, let me try it out and see 😃

---

So returning the error is kinda obstructing. Since the `ListDatabases()` only return the database you initially connected to, you can't really change anyway right now. I would suggest to keep the `no-op`. Thoughts?

---

aha, ok gotcha!
Hmm I am tempted to say we should wait implementing these methods in a separate PR,  then to actually implement them with no-op. Given, we don't want to confuse a user with feature of seeing databases but they aren't seeing what they 'expect'. Then it make sense to show nothing at all (because the feature doesn't exist) 😄 

Do you think we'll make a PR adding this before the v0.2.0 release? 😋 if so, we can add this code and update it accordingly

---

So after not to much looking in to this, one can attach to new/more databases. Then you can run `use other_db` and swap between them. But I would say that this is not really how one typically use DuckDB. But I liked the visual "feedback" of seeing what database I was attached to. But if it's possible to disable the select action only, that would maybe be a hybrid solution?

Regardless, I think you're right that this can be it's own PR after `v0.2.0` 😃 So, lets do that!